yb-voyager/cmd/metadataDB.go

```diff
@@ -97,3 +97,35 @@ func truncateTablesInMetaDb(exportDir string, tableNames []string) error {
 	}
 	return nil
 }
+
+// =====================================================================================================================
+
+type MetaDB struct {
+	db *sql.DB
+}
+
+func NewMetaDB(exportDir string) (*MetaDB, error) {
+	db, err := sql.Open("sqlite3", getMetaDBPath(exportDir))
+	if err != nil {
+		return nil, fmt.Errorf("error while opening meta db :%w", err)
+	}
+	return &MetaDB{db: db}, nil
+}
+
+func (m *MetaDB) MarkSegmentAsProcessed(segmentNum int64) error {
+	var query string
+	if importDestination == TARGET_DB {
+		query = fmt.Sprintf(`UPDATE %s SET imported_in_targetdb = 1 WHERE segment_no = %d;`, QUEUE_SEGMENT_META_TABLE_NAME, segmentNum)
+	} else if importDestination == FF_DB {
+		query = fmt.Sprintf(`UPDATE %s SET imported_in_ffdb = 1 WHERE segment_no = %d;`, QUEUE_SEGMENT_META_TABLE_NAME, segmentNum)
+	} else {
+		return fmt.Errorf("invalid importer type: %s", importDestination)
+	}
+
+	_, err := m.db.Exec(query)
```

Comment:
Just as a safety, it would be good to assert that updatedRows = 1 when we run updates.

Reply:
Yep added